### PR TITLE
EFA RDMA read

### PR DESCRIPTION
**Pull request by galpress**
Support RDMA operations through the new verbs send API and report RDMA capabilities through the query device direct verb.

### CODE REVIEW

**Comment by jgunthorpe** [comment]
There are a couple checkpatch errors that should be fixed

File: providers/efa/man/efadv_query_device.3.md [hunk 1]
```diff
@@ -61,6 +63,13 @@ struct efadv_device_attr {
 *inline_buf_size*
 :	Maximum inline buffer size.
 
+*max_wr_rdma_sge*
```
Comment (jgunthorpe) [inline providers/efa/man/efadv_query_device.3.md]: Isn't this a standard query already?

Comment (galpress) [reply providers/efa/man/efadv_query_device.3.md]: According to the documentation:

>         int                     max_sge_rd;             /* Maximum number of s/g per WR for RD QPs */


This isn't quite an RD QP.

**Comment by galpress** [comment]
> There are a couple checkpatch errors that should be fixed

The two occurrences are line break warnings (a bit over 80), which I think will hurt the readability.
Your call if you prefer the line break 😃

File: providers/efa/man/efadv_query_device.3.md (hunk 1 above)
Comment (jgunthorpe) [reply providers/efa/man/efadv_query_device.3.md]: max_sge is the max number of SGEs usable for all QPs types for everything but RDMA READ
max_sge_rd is the max number of SGEs usable for all QP types for RDMA READ

Where did you find that wrong comment?

**Comment by jgunthorpe** [comment]
Those are just function decls, it is expected to wrap them. Touch the lines with clang-format and it will do a reasonable job.

File: providers/efa/man/efadv_query_device.3.md (hunk 1 above)
Comment (galpress) [reply providers/efa/man/efadv_query_device.3.md]: Ah, shouldn't have trust anything but the man in the repo..
https://linux.die.net/man/3/ibv_query_device

Thanks, I will remove this field.

**Comment by galpress** [comment]
> Those are just function decls, it is expected to wrap them. Touch the lines with clang-format and it will do a reasonable job.

Will do.

**Comment by galpress** [comment]
Added RDMA read support report to the DV.

**Comment by galpress** [comment]
Is this ready to merge?

**Comment by rleon** [comment]
Gal, please update first patch and I'll merge it.

Thanks

**Comment by galpress** [comment]
I assume you meant the second patch (kernel headers), done.
Thanks Leon

**Pull request merged**